src/test/java/com/pretz/dyntreeserver/controller/DynTreeValidationTest.java

```diff
@@ -0,0 +1,271 @@
+package com.pretz.dyntreeserver.controller;
+
+import com.pretz.dyntreeserver.service.dto.DynTreeDTO;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+import javax.validation.ValidatorFactory;
+import java.time.Year;
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class DynTreeValidationTest {
+
+    private DynTreeDTO testDynTreeDTO;
+    private static Validator validator;
+
+    @BeforeAll
+    public static void createValidator() {
+        ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+        validator = validatorFactory.getValidator();
+    }
+
+    @Test
+    public void shouldPassValidation() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName("Duda")
+                .familyCount(10)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
+
+        Set<ConstraintViolation<DynTreeDTO>> violations
+                = validator.validate(testDynTreeDTO);
+
+        assertTrue(violations.isEmpty());
+    }
+
+    @Test
+    public void shouldFailValidationBlankFamilyName() {
+        testDynTreeDTO = DynTreeDTO.builder()
+                .familyName("    ")
+                .familyCount(10)
+                .generationsCount(3)
+                .childrenPerCharacter(2.2)
+                .nameListId(4L)
+                .maxAge(90)
+                .maturityAge(18)
+                .startingYear(Year.of(1950))
+                .mainCharacterName("Andrzej")
+                .build();
```

Comment:
you're repeating most of these fields in each test, maybe you could add a default builder and just overwrite fields you need to?

Reply:
I've meant that instead of copypasting all these stuff around, you could initialize the `testDynTreeDTO` in a `setup` or `beforeAll` (I don't remember what are the hooks here) function and inside each test just write:
```
testDynTreeDTO
  .fieldThatActuallyChanges(value)
  .build();
```

Reply:
Well, builder initializes object in one step. I cannot build and already built object. xD If I would use it in this way, the other fields will be null and therefore the object would fail the validation by side-effects.

Reply:
More explicit explanation:
```
setup() {
testDynTreeDTO = DynTreeDTO.builder()
                .familyName("    ")
                .familyCount(10)
                .generationsCount(3)
                .childrenPerCharacter(2.2)
                .nameListId(4L)
                .maxAge(90)
                .maturityAge(18)
                .startingYear(Year.of(1950))
                .mainCharacterName("Andrzej");


testFooBar() {
  DynTreeDTO dynTreeDTO = testDynTreeDTO.setValueFoo("bar").build;
}
```